project/cmake/addons/bootstrap/CMakeLists.txt

```diff
@@ -0,0 +1,61 @@
+project(kodi-addons-bootstrap)
+
+cmake_minimum_required(VERSION 2.8)
+
+list(APPEND CMAKE_MODULE_PATH ${PROJECT_SOURCE_DIR})
+
+# make sure CMAKE_INSTALL_PREFIX is properly set
+if(CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT OR NOT CMAKE_INSTALL_PREFIX)
+  set(CMAKE_INSTALL_PREFIX "${PROJECT_SOURCE_DIR}/../addons")
+endif()
+list(APPEND CMAKE_PREFIX_PATH ${CMAKE_INSTALL_PREFIX})
+
+# figure out where the build directory is located
+if(NOT BUILD_DIR)
+  set(BUILD_DIR "${CMAKE_BINARY_DIR}/build")
+else()
+  file(TO_CMAKE_PATH "${BUILD_DIR}" BUILD_DIR)
+endif()
+get_filename_component(BUILD_DIR "${BUILD_DIR}" ABSOLUTE)
+
+# figure out which addons to bootstrap (defaults to all)
+if(NOT ADDONS_TO_BUILD)
+  set(ADDONS_TO_BUILD "all")
+  message(STATUS "Bootstrapping all addons")
+else()
+  message(STATUS "Bootstrapping following addons: ${ADDONS_TO_BUILD}")
+endif()
+
+# make sure that the addon type has been specified
+if(NOT ADDON_TYPE)
+  message(FATAL_ERROR "No addon type specified")
+endif()
+
+# make sure that the addon definitions repository has been specified
+if(NOT ADDON_DEFINITIONS_REPOSITORY)
+  message(FATAL_ERROR "No addons definition repository specified")
```

Comment:
IMO we need to support bootstrapping being executed multiple times for different meta repositories (or adjust it to be able to handle multiple meta repositories which are defined somewhere) independent of whether we ourselves will use it. And that means that we need to choose a name to be used in `externalproject_add()`. I'm fine with adding default values for `ADDON_TYPE` (e.g. `all`) and `ADDON_DEFINITIONS_REPOSITORY` but if someone wants to bootstrap multiple meta repositories and then build addons from both meta repositories we need to be able to choose different names to be used in `externalproject_add()`.

The reason why I would prefer multiple meta repositories is that it would make it easier to e.g. only build all PVR addons (if something has changed in the PVR API) without having to manually define all the PVR addons that need to be built. You'd just call this bootstrap buildsystem with the meta repo containing all PVR addon definitions and you'd be good to go.